runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillStateCache.java

```diff
@@ -225,24 +225,23 @@ public long getWeight() {
 
   /** Entry in the state cache that stores a map of values. */
   private static class StateCacheEntry implements Weighted {
-    private final HashMap<NamespacedTag<?>, WeightedValue<?>> values;
+    private final IdentityHashMap<InternedByteString, WeightedValue<?>> values;
```

Comment:
if we in some cases are not using a consistent WindmillStateTagUtil then we could have InternedByteStrings which are equivalent but not the same object.